appender.go

```diff
@@ -326,21 +330,19 @@ func (a *Appender) flush(ctx context.Context, bulkIndexer *bulkIndexer) error {
 			if info.Status >= 500 {
 				serverFailed++
 			}
-			// NOTE(axw) error type and reason are included
-			// in the error message so we can observe different
-			// error types/reasons when logging is rate limited.
```

Comment:
I think the NOTE is still relevant - log messages may still be duplicated across multiple requests, and a rate limited logger could still be used to suppress them.

What is the benefit to this change vs. using a rate-limited logger like https://pkg.go.dev/go.uber.org/zap@v1.26.0/zapcore#NewSamplerWithOptions?

Reply:
What I like about this approach is that it quantifies the number of documents in the log message, and that's included in the log line.